mlir/test/Dialect/Linalg/roundtrip.mlir

```diff
@@ -485,6 +485,48 @@ func.func @variadic_reduce_memref(%input1: memref<16x32x64xf32>,
 
 // -----
 
+func.func @reduce_asymmetric(%input: tensor<16x32x64xi32>, %input2: tensor<16x32x64xi32>,
+                  %init: tensor<16x64xi32>) -> tensor<16x64xi32> {
+  %reduce = linalg.reduce
```

Comment:
Like I dont understand what this is trying to do. You are effectively doing a multiply and a reduce? How is this different from just using a `linalg.generic`. As in why use a `linalg.reduce` better than `linalg.generic`?

Reply:
Yes this is a multiply-accumulate type operation. The difference is just that it's higher-level than a linalg generic. That is nice when you want to generate this op in a compiler pass, or when you want to pattern-match it. All linalg ops are just sugar over a generic, so I don't think this use case should be disqualified because it can be implemented with a generic.